backend/src/models/secretApproval.ts

```diff
@@ -0,0 +1,44 @@
+import { Schema, Types, model } from "mongoose";
+
+export interface ISecretApproval {
+  _id: Types.ObjectId;
+  workspace: Types.ObjectId;
+  environment: string;
+  secretPath?: string;
+  approvers: Types.ObjectId[];
+  approvals: number;
+}
+
+const secretApprovalSchema = new Schema<ISecretApproval>(
+  {
+    workspace: {
+      type: Schema.Types.ObjectId,
+      ref: "Workspace",
+      required: true
+    },
+    approvers: [
+      {
+        // user associated with the personal secret
+        type: Schema.Types.ObjectId,
+        ref: "Membership"
```

Comment:
curious why you choose `Membership`. I would think approvers are users. I also don't see Membership of secretApprovalSchema being used anywhere yet

Reply:
It should be membership because using user id is a idea.

When user leaves org membership  will be also removed. If its user id based even when user leaves they will be able to see it or checks would be complicated.

Reply:
makes sense